x/emissions/module/rewards/rewards_internal.go

```diff
@@ -528,6 +630,29 @@ func GetAllReputersOutput(
 			newCoefficients[j] = min
 		}
 
+		// Normalize the new coefficients
+		coeffSum, err := alloraMath.SumDecSlice(newCoefficients)
+		if err != nil {
+			return nil, nil, errors.Wrap(err, "GetAllReputersOutput, err summing new coefficients")
+		}
+
+		// Avoid division by zero if sum is zero (shouldn't happen with EpsilonDec clamp, but safety first)
+		if coeffSum.IsZero() || coeffSum.Lt(alloraMath.EpsilonDec()) {
+			// If the sum is effectively zero even after epsilon clamp, something is wrong. Maybe return initial state or error.
+			// For now, let's return an error, as this state indicates a potential deeper issue.
+			return nil, nil, errors.Wrap(types.ErrDivisionByZero, "GetAllReputersOutput, coefficient sum effectively zero during normalization")
+		}
+
+		normalizedCoefficients := make([]alloraMath.Dec, len(newCoefficients))
+		for j := range newCoefficients {
+			normalizedCoefficients[j], err = newCoefficients[j].Quo(coeffSum)
+			if err != nil {
+				return nil, nil, errors.Wrapf(err, "GetAllReputersOutput, error normalizing coefficient %d", j)
+			}
+		}
+		// Update newCoefficients to use the normalized values for subsequent checks
+		newCoefficients = normalizedCoefficients
```

Comment:
```suggestion
		for j := range newCoefficients {
			newCoefficients[j], err = newCoefficients[j].Quo(coeffSum)
			if err != nil {
				return nil, nil, errors.Wrapf(err, "GetAllReputersOutput, error normalizing coefficient %d", j)
			}
		}
```

I guess we can reuse the same array instead of allocating a new one